app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,47 @@
+class PostsController < ApplicationController
+
+  def index
+    @posts = Post.all
+  end
+
+  def new
+    authenticate_user!
```

Comment:
You can make your code DRYer by using `before_action` to run `authenticate_user!` (vs. writing it out for each controller action).